packages/core/nuxt-theme-module/theme/pages/MyAccount.vue

```diff
@@ -68,6 +67,17 @@ export default {
     OrderHistory,
     MyReviews
   },
+
```

Comment:
what about this:
https://github.com/DivanteLtd/vue-storefront/blob/next/packages/commercetools/theme/middleware/checkout.js

I think you can do something similar and we can have just one middleware in the CT theme, it needs a bit refactoring, but both require user so:

- rename checkout.js to the eg. access-guard.js
- move checkout logic to the separate file or just function above
- implement the logic for my account section in another function
- use both in the middleware

in this case, you will benefit from the user call which actually is being called even now

Reply:
Currently this middleware is part of the core, meaning that it works in all implementations. If we move it to CT, other integrations will lose it, unless we repeat it for every one of them.